paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentSelection.kt

```diff
@@ -392,7 +383,7 @@ internal val PaymentSelection.paymentMethodType: String
         PaymentSelection.Link -> "link"
         is PaymentSelection.New.Card -> paymentMethodCreateParams.typeCode
         is PaymentSelection.New.GenericPaymentMethod -> paymentMethodCreateParams.typeCode
-        is PaymentSelection.New.LinkInline -> linkPaymentDetails.paymentMethodCreateParams.typeCode
+        is PaymentSelection.New.LinkInline -> paymentMethodCreateParams.typeCode
```

Comment:
You should be able to just have one `PaymentSelection.New` case, instead of multiple `PaymentSelection.New.*` cases.